File: app/src/main/java/com/silentgames/silent_planet/model/fractions/factionType/Humans.java
package com.silentgames.silent_planet.model.fractions.factionType;

import com.silentgames.silent_planet.model.fractions.Fractions;
import com.silentgames.silent_planet.model.fractions.FractionsEnum;

/**
 * Created by gidroshvandel on 27.09.16.
 */
public class Humans extends Fractions {

    private static Humans instance;

    public static synchronized Humans getInstance(){
        if(instance == null){
            instance = new Humans();
        }
        return instance;
    }

    private Humans() {
        this.setFractionsEnum(FractionsEnum.Humans);
    }
}

File: app/src/main/java/com/silentgames/silent_planet/model/fractions/factionType/Humans.kt
package com.silentgames.silent_planet.model.fractions.factionType

import com.silentgames.silent_planet.model.fractions.Fractions
import com.silentgames.silent_planet.model.fractions.FractionsType

/**
 * Created by gidroshvandel on 27.09.16.
 */
class Humans private constructor() : Fractions(FractionsType.HUMAN) {

    companion object {

        private var instance: Humans? = null

        @Synchronized
        fun getInstance(): Humans {
            if (instance == null) {
                instance = Humans()
            }
            return instance!!
        }
    }
}
